xs2a-adapter-aspsp-registry/src/main/java/de/adorsys/xs2a/adapter/registry/AspspSearchServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package de.adorsys.xs2a.adapter.registry;
+
+import de.adorsys.xs2a.adapter.service.AspspRepository;
+import de.adorsys.xs2a.adapter.service.AspspSearchService;
+import de.adorsys.xs2a.adapter.service.model.Aspsp;
+import org.iban4j.Iban;
+
+import java.util.List;
+import java.util.Optional;
+
+public class AspspSearchServiceImpl implements AspspSearchService {
+
+    private final AspspRepository luceneAspspRepository;
+
+    public AspspSearchServiceImpl(AspspRepository luceneAspspRepository) {
+        this.luceneAspspRepository = luceneAspspRepository;
+    }
+
+    @Override
+    public List<Aspsp> findByIban(String iban, String after, int size) {
+        String bankCode = Iban.valueOf(iban).getBankCode();
```

Comment:
If `iban` format is not correct, `Iban.valueOf(iban)` will throw one of the iban4j custom exceptions. As we have no strategy for handling them, they will be handled as `Exception`, so the client will get the `500 INTERNAL SERVER ERROR` response. Is it the expected behavior?

Reply:
Good point. Added a new exception and handler.